services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusController.kt

```diff
@@ -80,4 +89,108 @@ class HakemusController(
         logger.info { "Found ${response.applications.size} applications for hanke $hankeTunnus" }
         return response
     }
+
+    @PutMapping("/hakemukset/{id}")
+    @Operation(
+        summary = "Update an application",
+        description =
+            """Returns the updated application.
+               The application can be updated until it has been sent to Allu.
+               If the application hasn't changed since the last update, nothing more is done.
+               The pendingOnClient value can't be changed with this endpoint.
+               Use [POST /hakemukset/{id}/send-application](#/application-controller/sendApplication) for that.
+            """
+    )
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(description = "The updated application", responseCode = "200"),
+                ApiResponse(
+                    description = "Request contains invalid data",
+                    responseCode = "400",
+                    content = [Content(schema = Schema(implementation = HankeErrorDetail::class))]
+                ),
+                ApiResponse(
+                    description = "An application was not found with the given id",
+                    responseCode = "404",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+                ApiResponse(
+                    description =
+                        "The application can't be updated because it has been sent to Allu",
+                    responseCode = "409",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+                ApiResponse(
+                    description = "Request contains non-compatible data",
+                    responseCode = "400",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+                ApiResponse(
+                    description = "Request contains invalid customer or contact data",
+                    responseCode = "400",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
```

Comment:
OpenAPI can have only one response per response code. These two will not show up in the output.

Reply:
Learned a new thing again, thank you! I removed duplicate 400 responses but added schema and example for HankeError and HankeErrorDetail in remaining 400 response.